### PR TITLE
Require and check for git

### PR DESCRIPTION
**Pull request by cgrant**

### CODE REVIEW

File: devai-cli/src/devai/commands/__init__.py [hunk 1]
```diff
@@ -1 +1,9 @@
+import shutil, os, sys
+
+
+os.environ["GIT_PYTHON_REFRESH"] = "quiet"
+if shutil.which("git") is None:
+    print("Git is required but not installed. Exiting.")
+    sys.exit(1)
```
Comment (gitrey) [inline devai-cli/src/devai/commands/__init__.py:7]: does it make sense to stop the run if git is missing?
in my case I'm not running the rag/load command.
all the other commands, review, jira, gitlab work fine even if git is missing.

can we add instructions to install it or whatever is required to make it work?

Comment (cgrant) [reply devai-cli/src/devai/commands/__init__.py:7]: The dependency does it's own check for git before the program runs. We'd have to do a nonstandard coding to make it conditional. Adding the check in the package to provide better user output vs the trace dump.

Comment (gitrey) [reply devai-cli/src/devai/commands/__init__.py:7]: This is how we have JIRA and GitLab functionality in the repo but not activated until we are ready to use it, have all the env vars set, etc.

https://github.com/GoogleCloudPlatform/genai-for-developers/blob/main/devai-cli/src/devai/commands/review.py#L60-L63

**Review by gitrey — APPROVED** [review]

**Comment by gitrey** [comment]
@cgrant can't finish merging - This branch is out-of-date with the base branch. ^

**Pull request merged**